lib/queue.rb

```diff
@@ -1,31 +1,65 @@
+require_relative 'linked_list'
+
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = @back = -1
+    @counter = 0
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+    end
+
+    # if @front == @back
+    #   raise NoSpaceError
+    # end
+
+    @counter += 1
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    #Decrease front
+    if @front == -1 
+      raise QuequeEmptyError
+    end
+
+    @counter -= 1
+    data = @store[@front] 
+    @store[@front] = nil
+
+    if @front == @back
+      @front = -1
+      @back = -1
+    elsif @front == (@store.length - 1)
```

Comment:
You should check to see if the queue is empty AFTER advancing `@front` along the buffer.  Basically move front to the next element of th enqueue and then check to see if front and back are equal.